Modules/Retailer/Http/Services/RetailerService.php

```diff
@@ -0,0 +1,256 @@
+            })
+            ->addColumn('store_type', function ($row) {
+                return $row->retailer->store_type;
+            })
+            ->addColumn('action', function ($row) {
+                return '<a href="' . url('/retailer/details/' . $row->id) . '"><span class="fas fa-eye"></span></a>  <a href="' . url('/order/retailer-order/' . $row->id) . '"><span class="fas fa-file-invoice-dollar"></span></a>  <a href="javascript:void(0)" onclick="event.preventDefault(); confirmDelete(' . $row->id . ');"><span class="fas fa-trash"></span></a> ';
+            })
+            ->rawColumns(['action'])
+            ->make(true);
+        $resultData = $result->getData(true);
+        $resultData['recordsFiltered'] = $filteredCount;
+        $resultData['draw'] = $request->draw;
+
+        return $resultData;
+    }
+
+    /**
+     * Details of retailer
+     *
+     * @param $id
+     * @return mixed
+     */
+    public function detail($id): mixed
+    {
+        $userRetailer = User::with('retailer')->findOrFail($id);
+        $detail = $userRetailer->retailer;
+        $country = Country::where('id', $userRetailer->retailer->country)->first();
+        $state = State::where('id', $userRetailer->retailer->state)->first();
+        $town = City::where('id', $userRetailer->retailer->town)->first();
+        $detail['name'] = $userRetailer->first_name . ' ' . $userRetailer->last_name;
+        $detail['email'] = $userRetailer->email ?? '';
+        $detail['country'] = $country->name ?? '';
+        $detail['state'] = $state->name ?? '';
+        $detail['town'] = $town->name ?? '';
+        $detail['shipping'] = $userRetailer->shippings;
+
+        return $detail;
+    }
+
+    /**
+     * Show Brand in Website
+     *
+     * @param $id
+     * @param $status
+     * @return array
+     */
+    public function changeStatus($id, $status): array
+    {
+        $user = User::find($id);
+        $user->active = $status;
+        $user->save();
+
+        return [
+            'res' => true,
+            'msg' => '',
+            'data' => ''
+        ];
+    }
+
+    /**
+     * Delete Retailer
+     *
+     * @param $id
+     * @return array
+     */
+    public function delete($id): array
+    {
+        $user = User::findOrFail($id);
+        $user->delete();
+        return [
+            'res' => true,
+            'msg' => 'Successfully deleted',
+            'data' => ''
+        ];
+    }
+
+    /**
+     * Get all country
+     *
+     * @return array
+     */
+    public function getCountry(): array
+    {
+        $countries = Country::select('id', 'shortname', 'name', 'phonecode')->get();
+        $data = array();
+        foreach ($countries as $country) {
+            $data[] = array(
+                'id' => $country->id,
+                'country_code' => $country->shortname,
+                'country_name' => $country->name,
+                'phone_code' => $country->phonecode
+            );
+        }
```

Comment:
Update this code with this one
$countries = Country::pluck('id', 'shortname', 'name', 'phonecode');

$data = $countries->map(function ($country) {
    return [
        'id' => $country['id'],
        'country_code' => $country['shortname'],
        'country_name' => $country['name'],
        'phone_code' => $country['phonecode']
    ];
})->toArray();

Reply:
Pluck is not working for multiple columns..that's why i added below code. please tell me if  wrong

$countries = Country::select('id', 'shortname', 'name', 'phonecode')->get();
        $data = $countries->map(function ($item, $key) {
            return [
                'id' => $item->id,
                'country_code' => $item->shortname,
                'country_name' => $item->name,
                'phone_code' => $item->phonecode,
            ];
        })->toArray();

Reply:
that's good.